crates/js-component-bindgen/src/ts_bindgen.rs

```diff
@@ -254,7 +254,7 @@ pub fn ts_bindgen(
                         instantiateCore?: (module: WebAssembly.Module, imports: Record<string, any>) => WebAssembly.Instance
                     ): {camel};
                     export function instantiate(
```

Comment:
Not an expert in TS, I'm not sure why we need 2 function signatures here (this one and the above). Should one be enough? And should just `Promise` be enough?

```typescript
                    export function instantiate(
                        getCoreModule: (path: string) => Promise<WebAssembly.Module>,
                        imports: ImportObject,
                        instantiateCore?: (module: WebAssembly.Module, imports: Record<string, any>) =>  Promise<WebAssembly.Instance>
                    ): Promise<{camel}>;
```

Reply:
It's an overload of the function in the fully sync case to note that if both functions are sync the return value is always known synchronously.

So I think this PR is correct but thanks for checking on this.